### PR TITLE
feat: Add toolbar for fragments + animation

### PR DESCRIPTION
**Pull request by anhanh11001**
Details:
- Add toolbar when scrolling in fragments
- Add animation to show and hide toolbar

Fixes: #1951 

Screenshots for the change:
ezgif-2-668b37a4a56f 
 ezgif-2-977580aedee5

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt [hunk 1]
```diff
@@ -184,6 +196,40 @@ class EventsFragment : Fragment(), BottomIconDoubleClick {
             onFavFabClick = favFabClickListener
             onHashtagClick = hashTagClickListener
         }
+
+        rootView.scrollView.setOnScrollChangeListener { _: NestedScrollView?, _: Int, scrollY: Int, _: Int, _: Int ->
+            if (scrollY > rootView.locationTextView.y + rootView.locationTextView.height) {
+                if (!rootView.toolbarLayout.isVisible) showToolbar()
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: these 2 if statements can be combined

File: app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt [hunk 2]
```diff
@@ -184,6 +196,40 @@ class EventsFragment : Fragment(), BottomIconDoubleClick {
...
+            } else {
+                if (rootView.toolbarLayout.isVisible) hideToolbar()
```
Comment (nikit19) [inline app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: `else if`

File: app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt (hunk 1 above)
Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: fixed

File: app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt (hunk 2 above)
Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: fixed

File: app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt [hunk 3]
```diff
@@ -157,6 +166,9 @@ class EventsFragment : Fragment(), BottomIconDoubleClick {
         rootView.notification.setOnClickListener {
             findNavController(rootView).navigate(EventsFragmentDirections.actionEventsToNotification())
         }
+        rootView.notificationToolbar.setOnClickListener {
+            findNavController(rootView).navigate(EventsFragmentDirections.actionEventsToNotification())
```
Comment (liveHarshit) [inline app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: Duplication can be removed

Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: notification is an ImageView shown when toolbar is hide, notificationToolbar is an icon in the toolbar when toolbar is shown, so it's not duplicate

Comment (liveHarshit) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: oh, that's not what I mean. I'm talking about `findNavController(rootView).navigate(EventsFragmentDirections.actionEventsToNotification())`. But it is okay. :+1:

Comment (anhanh11001) [reply app/src/main/java/org/fossasia/openevent/general/event/EventsFragment.kt]: oh okay, I'm fixing now, besides, build is failing so I need to make some fixes as well

**Pull request merged**